ProcessLib/HeatTransportBHE/LocalAssemblers/HeatTransportBHELocalAssemblerSoil-impl.h

```diff
@@ -114,16 +120,69 @@ void HeatTransportBHELocalAssemblerSoil<ShapeFunction, IntegrationMethod>::
                     MaterialPropertyLib::PropertyType::specific_heat_capacity)
                 .template value<double>(vars, pos, t);
 
-        auto const density_s =
-            solid_phase.property(MaterialPropertyLib::PropertyType::density)
+        auto const density_f =
+            liquid_phase.property(MaterialPropertyLib::PropertyType::density)
                 .template value<double>(vars, pos, t);
-        // for now only using the solid phase parameters
+
+        auto const heat_capacity_f =
+            liquid_phase
+                .property(
+                    MaterialPropertyLib::PropertyType::specific_heat_capacity)
+                .template value<double>(vars, pos, t);
+
+        auto const porosity =
+            medium.property(MaterialPropertyLib::PropertyType::porosity)
+                .template value<double>(vars, pos, t);
+
+        auto const velocity_arr =
+            liquid_phase
+                .property(MaterialPropertyLib::PropertyType::phase_velocity)
+                .value(vars, pos, t);
+        auto velocity = Eigen::Map<Eigen::Matrix<double, 1, 3> const>(
```

Comment:
```suggestion
        auto const velocity = Eigen::Map<Eigen::Vector3d const>(
```

Reply:
Agreed. The suggestion has been committed.